src/libsyntax/feature_gate.rs

```diff
@@ -290,6 +290,12 @@ declare_features! (
     // The `repr(i128)` annotation for enums.
     (active, repr128, "1.16.0", Some(35118), None),
 
+    // Allows the use of `#[ffi_pure]` on foreign functions.
+    (active, ffi_pure, "1.34.0", Some(58329), None),
+
+    // Allows the use of `#[ffi_const]` on foreign functions.
+    (active, ffi_const, "1.34.0", Some(58328), None),
```

Comment:
Consider putting descriptions of these into the unstable book.

(Since they're described as "C attributes" I went looking in https://en.cppreference.com/w/cpp/language/attributes#Standard_attributes to see what they meant, but they weren't defined there.)